rules/msk_topic_config.go

```diff
@@ -80,9 +71,49 @@ func (r *MSKTopicConfigRule) validateTopicConfig(runner tflint.Runner, topic *hc
 		return err
 	}
 
+	configAttr, hasConfig := topic.Body.Attributes["config"]
+	if !hasConfig {
+		err := runner.EmitIssue(
+			r,
+			"missing config attribute: the topic configuration must be specified in a config attribute",
+			topic.DefRange,
+		)
+		if err != nil {
+			return fmt.Errorf("emitting issue: missing config block: %w", err)
+		}
+		return nil
+	}
+
+	/* construct a mapping between the config key and the config KeyPair. This helps in both checking if a key is defined and to propose fixes to the values*/
+	configKeyToPairMap, err := constructConfigKeyToPairMap(configAttr)
+	if err != nil {
+		return err
+	}
+
+	if err := r.validateCompressionType(runner, configAttr, configKeyToPairMap); err != nil {
+		return err
+	}
```

Comment:
This could probably be a separate function.

But then I realised maybe naming needs an update since this `validateTopicConfig` function is actually validating more than just the config of the `kafka_topic` resource (since e.g. the `replication_factor` is not under `kafka_topic.config`)

```suggestion
	if err := r.validateTopicConfig(runner, topic); err != nil { // validateTopicConfig is validating _just_ `kafka_topic.config`
		return err
	}

	return nil
```

Reply:
I'll leave it as is for now, as I consider the replication factor belonging to the topic configuration too, but it's an outspoke configuration as it is a required attribute in the terraform plugin.